corehq/apps/hqadmin/management/commands/stale_data_in_es.py

```diff
@@ -181,30 +181,30 @@ def run(cls, run_config):
                 all_domains_with_dedicated_search_index = [run_config.domain]
 
         case_query = CaseES(for_export=True)
-        for chunk in cls.get_sql_chunks(all_domains, run_config):
+        for chunk in cls.get_sql_chunks(all_domains, run_config, 'case'):
             yield from cls._yield_missing_in_es(chunk, case_query)
 
         # USH-4684: The yield from these for loop can be moved in the one for CaseES
         case_search_query = CaseSearchES(for_export=True)
         for domain in all_domains_with_search_index:
-            for chunk in cls.get_sql_chunks(domain, run_config):
+            for chunk in cls.get_sql_chunks(domain, run_config, f'case-search-{domain}'):
                 yield from cls._yield_missing_in_es(chunk, case_search_query)
 
         for domain in all_domains_with_dedicated_search_index:
             case_search_index = settings.CASE_SEARCH_SUB_INDICES[domain]['index_cname']
-            case_search_query = CaseSearchES(for_export=True, index=case_search_index)
```

Comment:
This should work - it's what we do for case search: https://github.com/dimagi/commcare-hq/blob/29b3ff8a9890b4c4b832cb7974e1281c00202e5d/corehq/apps/case_search/utils.py#L181-L185
what was the error you were getting?

Reply:
It was just not updating the index. Further down I was using it in `es_base_query.index` and if I set the index in the constructor I get "case_search" back instead of "case_search_bha"

Reply:
Ah, I see - it doesn't override the `index` property, but it does set the adapter appropriately:
```python
>>> query = CaseSearchES(index='case_search_bha')
>>> query.index
'case_search'
>>> query.adapter.canonical_name
'case_search_bha'
```
So queries should go to the new index correctly, even though `query.index` isn't updated.  I'm not sure why we don't also set `index` there, that seems confusing/wrong.